src/modules/endpoint/reducers/pushPoi.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import type { CityContentStateType, PoiRouteStateType } from '../../app/StateType'
+import type { PushPoiActionType } from '../../app/StoreActionType'
+import { PoiModel } from '@integreat-app/integreat-api-client'
+
+const pushPoi = (state: CityContentStateType, action: PushPoiActionType): CityContentStateType => {
+  const { pois, path, key, language, resourceCache, cityLanguages, city } = action.params
+
+  if (!key) {
+    throw new Error('You need to specify a key!')
+  }
+
+  const getPoiRoute = (): PoiRouteStateType => {
+    const allAvailableLanguages = new Map(cityLanguages.map(lng => [lng.code, null]))
```

Comment:
Here is the reason why the `availableLanguages` state has a `?string`. Is this required here? Can we link to pois in different languages? @Schedulaar do you know more here?

Reply:
We should be able to do so. As for the events, this should only filled with `null` if it corresponds to the list view.